build breaking type checker overflow: "overflow evaluating the requirement `&tokio_reactor::poll_evented::PollEvented<_>: std::io::Read`"

### Description

The [`master` build broke](https://circleci.com/gh/filecoin-project/rust-fil-proofs/22445) with the following compilation error: 

```
error[E0275]: overflow evaluating the requirement `&tokio_reactor::poll_evented::PollEvented<_>: std::io::Read`
  --> filecoin-proofs/benches/preprocessing.rs:61:5
   |
61 |     write_padded(&mut data[..].as_ref(), file).unwrap();
   |     ^^^^^^^^^^^^
   |
   = help: consider adding a `#![recursion_limit="128"]` attribute to your crate
   = note: required because of the requirements on the impl of `std::io::Read` for `&tokio_reactor::poll_evented::PollEvented<tokio_reactor::poll_evented::PollEvented<_>>`
   = note: required because of the requirements on the impl of `std::io::Read` for `&tokio_reactor::poll_evented::PollEvented<tokio_reactor::poll_evented::PollEvented<tokio_reactor::poll_evented::PollEvented<_>>>`
   = note: required because of the requirements on the impl of `std::io::Read` for `&tokio_reactor::poll_evented::PollEvented<tokio_reactor::poll_evented::PollEvented<tokio_reactor::poll_evented::PollEvented<tokio_reactor::poll_evented::PollEvented<_>>>>`
```

--- a/filecoin-proofs/benches/preprocessing.rs
+++ b/filecoin-proofs/benches/preprocessing.rs
@@ -49,15 +49,15 @@ fn preprocessing_benchmark(c: &mut Criterion) {
     );
 }
 
-fn write_padded_bench(file: &mut File, mut data: Vec<u8>) {
-    let _ = write_padded(&mut data[..].as_ref(), file).unwrap();
+fn write_padded_bench(file: &mut File, data: Vec<u8>) {
+    let _ = write_padded(&mut &data[..], file).unwrap();
     let padded_written = file.seek(SeekFrom::End(0)).unwrap() as usize;
 
     assert!(padded_written > data.len());
 }
 
-fn write_padded_unpadded_bench(file: &mut File, mut data: Vec<u8>) {
-    write_padded(&mut data[..].as_ref(), file).unwrap();
+fn write_padded_unpadded_bench(file: &mut File, data: Vec<u8>) {
+    write_padded(&mut &data[..], file).unwrap();
 
     let padded_written = file.seek(SeekFrom::End(0)).unwrap() as usize;
 

